File: Programmers/jsh/11주차/모음사전.py
#경우의수가 최대6^5이므로 다중for문을 사용해도 무방하므로
#모든경우를 계산하여 해결함
def solution(word):
    answer = 0
    lists=['A','E','I','O','U']
    for i in lists:
        temp=['x','x','x','x','x']
        temp[0]=i
        answer+=1
        if word=="".join(temp[:1]):
            return answer
        for j in lists:
            temp[1]=j
            answer+=1
            if word=="".join(temp[:2]):
                return answer
            for k in lists:
                temp[2]=k
                answer+=1
                if word=="".join(temp[:3]):
                    return answer
                for l in lists:
                    temp[3]=l
                    answer+=1
                    if word=="".join(temp[:4]):
                        return answer
                    for q in lists:
                        temp[4]=q
                        answer+=1
                        if word=="".join(temp[:5]):
                            return answer
                        
    return answer

File: Programmers/Segun/11주차/모음사전.cpp
//A서부터 UUUUU 까지 모든 나올수있는 경우의 수를 확인하고 주어진 word와 같은 문장이 나왔을때까지 진행

#include <string>
#include <vector>
#include<iostream>

using namespace std;

char alpha[5]={'A','E','I','O','U'};
string str="XXXXX";
int answer = 0;
bool chk;
void DFS(int cnt,string word){
    if(cnt>5||chk==true){return;}
    if(str.substr(0,cnt)==word){chk=true;return;}
    answer++;
    for(int i=0;i<5;i++){
        str[cnt]=alpha[i];
        DFS(cnt+1,word);
    }
}

int solution(string word) {
    DFS(0,word);
    return answer;
}
